server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithoutValueFilter.java

```diff
@@ -530,6 +560,10 @@ protected void fillCache(int seriesIndex) throws IOException, InterruptedExcepti
       synchronized (seriesReaderList.get(seriesIndex)) {
         // we only need to judge whether to submit another task when the queue is not full
         if (blockingQueueArray[seriesIndex].remainingCapacity() > 0) {
+          if (batchDataLengthList != null
+              && batchDataLengthList.get(seriesIndex) >= rowLimit + rowOffset) {
+            return;
+          }
```

Comment:
You can safely delete this `if` block. It seems that we don't need to do this judgement here, because we will set `hasRemaining` to false in sub thread.

Reply:
You are right~